src/ui/server/controllers/insurance/eligibility/companies-house-search/index.ts

```diff
@@ -64,51 +64,59 @@ export const get = (req: Request, res: Response) =>
  */
 export const post = async (req: Request, res: Response) => {
   try {
-    const payload = constructPayload(req.body, [FIELD_ID]);
+    const payload = constructPayload(req.body, [FIELD_ID]) as CompaniesHouseResponse;
 
     /**
      * Check that a companies house number has been provided.
      * If not, render the template with validation errors.
      */
-    const validationErrors = generateValidationErrors(payload);
+    const formValidationErrors = generateValidationErrors(payload);
 
-    if (validationErrors) {
+    if (formValidationErrors) {
       return res.render(TEMPLATE, {
         ...insuranceCorePageVariables({
           PAGE_CONTENT_STRINGS,
           BACK_LINK: req.headers.referer,
         }),
         userName: getUserNameFromSession(req.session.user),
         ...PAGE_VARIABLES,
-        validationErrors,
+        validationErrors: formValidationErrors,
+        submittedValues: req.body,
       });
     }
 
     /**
      * 1) Call companies house API (via our own API)
+     * 4) If validationErrors are returned, render the page with validation errors.
      * 2) If apiError is returned, redirect to COMPANIES_HOUSE_UNAVAILABLE.
-     * 3) If validationErrors are returned, render the page with validation errors.
+     * 3) If the company is not active, redirect to COMPANY_NOT_ACTIVE.
+     * 4) If validationErrors are returned, render the page with validation errors.
      */
     const response = await companiesHouse.search(payload);
 
-    if (response.apiError) {
-      return res.redirect(COMPANIES_HOUSE_UNAVAILABLE);
-    }
-
-    if (response.validationErrors) {
+    if (response.notFound) {
       return res.render(TEMPLATE, {
         ...insuranceCorePageVariables({
           PAGE_CONTENT_STRINGS,
           BACK_LINK: req.headers.referer,
         }),
         userName: getUserNameFromSession(req.session.user),
         ...PAGE_VARIABLES,
-        validationErrors: response.validationErrors,
+        validationErrors: generateValidationErrors({}),
+        submittedValues: req.body,
```

Comment:
payload

Reply:
Updated

Reply:
has this been changed? I can't see it

Reply:
Just checked, yes it's been changed. Must be github cache, happens sometimes